src/Helpers/JwksVerifier.php

```diff
@@ -0,0 +1,64 @@
+<?php
+declare(strict_types=1);
+
+namespace Auth0\SDK\Helpers;
+
+use Auth0\SDK\Exception\InvalidTokenException;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Signer\Rsa\Sha256 as RsSigner;
+use Lcobucci\JWT\Token;
+
+/**
+ * Class JwksVerifier
+ *
+ * @package Auth0\SDK\Helpers
+ */
+final class JwksVerifier extends SignatureVerifier
+{
+
+    /**
+     * JWKS array with kid as keys, PEM cert as values.
+     *
+     * @var array
+     */
+    private $jwks;
+
+    /**
+     * JwksVerifier constructor.
+     *
+     * @param array $jwks JWKS to use.
+     */
+    public function __construct(array $jwks)
```

Comment:
Sample used a URL here but I don't think that HTTP requests and caching should be a part of this class, should be more focused on the task. See the code above to see how this would look in situ:

```php
if ('RS256' === $this->idTokenAlg) {
    $jwksFetcher = new JWKFetcher($this->cacheHandler, $this->guzzleOptions);
    $jwks        = $jwksFetcher->getKeys($this->idTokenIss.'.well-known/jwks.json');
    $sigVerifier = new JwksVerifier( $jwks );
} else if ('HS256' === $this->idTokenAlg) {
    $sigVerifier = new SymmetricVerifier($this->clientSecret);
}
```